### DISCUSSION TITLE
Docker Qnap, ZigBee2MQTT e dongle CC2531 non funziona

### DISCUSSION THREAD

**Discussion by gmarongiu71** [indomus/forum#1627] [Hardware & Firmware]
Tanto per iniziare saluto a tutti, mi sto addentrando in questo mondo e fin'ora non avuto problemi, tranne questo.
Ho acquistato un dongle USB CC2531 da inserire nella porta USB del mio QNAP TS-251A e di seguito i problemi.

Non riesco a far partire il servizio, driver installato gli output sembrano corretti ma non partono. Grazie a chi mi aiuta, premetto che ho anche un UPS collegato via USB che funziona senza problemi.
Il Nas è il TS-251A con OS 5.0
Allego screenshot:

Driver Installato
![Driver](https://user-images.githubusercontent.com/103220010/162620592-864a7938-f201-441d-8ee5-145d5c9e7cf4.png)

Querry via SSH con Putty
![VerUSB](https://user-images.githubusercontent.com/103220010/162620604-f1383839-a89d-427f-96b0-502f4c4c163c.png)

Docker ZigBee2MQTT che non parte
![non parte](https://user-images.githubusercontent.com/103220010/162620646-05fa9798-b98b-4a7e-bcba-749f174ed19a.png)

Config zigbee2mqtt:
![Config USB](https://user-images.githubusercontent.com/103220010/162620677-b112b65e-8a56-4aee-80ef-a821fea031ee.jpeg)

Config Mosquitto con ovviamente la password configurato in "pwfile":

persistence true
persistence_location /mosquitto/data
listener 1883
log_dest file /mosquitto/log/mosquitto.log
log_timestamp_format %Y-%m-%dT%H:%M:%S
allow_anonymous false
password_file /mosquitto/config/pwfile

**Comment by snakuzzo** [comment]
Ma quando hai creato il container hai mappato correttamente la porta USB? Da quello che vedo nel log non mi pare proprio...visto che non trova proprio `/dev/ttyUSB0`.

Devi...
1. Capire con certezza qual è la porta sull'host
2. Mapparla quando crei il container

**Reply by gmarongiu71** [reply]
Ho seguito la guida, e con la prova via ssh mi da due porte, la 0 e la 1 che ho provato entrambi nel file di configurazione. Infatti se metto 1 l'errore mi si sposta sulla TTYUSB1. Quando crei il container dove mi fa configurare la porta?
Grazie

**Reply by snakuzzo** [reply]
Rileggi attentamente la guida

**Reply by gmarongiu71** [reply]
Riprovo, ma il problema potrebbe essere nella configurazione di mosquitto o di ZigBee2MQTT? Nella seconda mi fa scegliere solo questo.
![Immagine 2022-04-10 171938](https://user-images.githubusercontent.com/103220010/162626625-5e909c4d-0d0e-4fd4-9ada-59117d939d46.png)


Questa è la mia configurazione prima di cliccare su crea.... Posso provare a spostare il dongle su tutte le porte del nas facendolo ripartire ogni volta

**Reply by gmarongiu71** [reply]
<html>
<body>
<div class="link-row" style="box-sizing: inherit; position: relative; display: flex; color: rgb(47, 47, 47); font-family: Roboto, Helvetica, Arial, sans-serif, Verdana, Tahoma, 微軟正黑體; font-size: 0px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(250, 250, 250); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><div class="data-row right even" style="box-sizing: inherit; overflow: hidden; display: flex; position: relative; background-color: rgb(255, 255, 255); -webkit-box-flex: 1; flex: 1 1 0%;"><div class="data" style="box-sizing: inherit; border-right: 0.0625rem solid rgb(200, 200, 200); border-top: none !important; border-bottom: none !important; border-left: 0.0625rem solid rgb(200, 200, 200); border-image: initial !important; width: 196.5px; line-height: 28px; font-size: 0.75rem; color: rgb(47, 47, 47); display: inline-block; padding-left: 0.625rem; padding-right: 0.625rem; overflow: hidden;">Nome</div><div class="data" style="box-sizing: inherit; border-right: 0.0625rem solid rgb(200, 200, 200); width: 589.5px; line-height: 28px; font-size: 0.75rem; color: rgb(47, 47, 47); display: inline-block; padding-left: 0.625rem; padding-right: 0.625rem; overflow: hidden;">zigbee2mqtt-1</div></div></div><div class="link-row" style="box-sizing: inherit; position: relative; display: flex; color: rgb(47, 47, 47); font-family: Roboto, Helvetica, Arial, sans-serif, Verdana, Tahoma, 微軟正黑體; font-size: 0px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(250, 250, 250); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><div class="data-row right even" style="box-sizing: inherit; overflow: hidden; display: flex; position: relative; background-color: rgb(244, 244, 244); -webkit-box-flex: 1; flex: 1 1 0%;"><div class="data" style="box-sizing: inherit; border-right: 0.0625rem solid rgb(200, 200, 200); border-top: none !important; border-bottom: none !important; border-left: 0.0625rem solid rgb(200, 200, 200); border-image: initial !important; width: 196.5px; line-height: 28px; font-size: 0.75rem; color: rgb(47, 47, 47); display: inline-block; padding-left: 0.625rem; padding-right: 0.625rem; overflow: hidden;">Comando</div><div class="data" style="box-sizing: inherit; border-right: 0.0625rem solid rgb(200, 200, 200); width: 589.5px; line-height: 28px; font-size: 0.75rem; color: rgb(47, 47, 47); display: inline-block; padding-left: 0.625rem; padding-right: 0.625rem; overflow: hidden;">/sbin/tini -- node index.js</div></div></div><div class="link-row" style="box-sizing: inherit; position: relative; display: flex; color: rgb(47, 47, 47); font-family: Roboto, Helvetica, Arial, sans-serif, Verdana, Tahoma, 微軟正黑體; font-size: 0px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(250, 250, 250); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><div class="data-row right even" style="box-sizing: inherit; overflow: hidden; display: flex; position: relative; background-color: rgb(255, 255, 255); -webkit-box-flex: 1; flex: 1 1 0%;"><div class="data" style="box-sizing: inherit; border-right: 0.0625rem solid rgb(200, 200, 200); border-top: none !important; border-bottom: none !important; border-left: 0.0625rem solid rgb(200, 200, 200); border-image: initial !important; width: 196.5px; line-height: 28px; font-size: 0.75rem; color: rgb(47, 47, 47); display: inline-block; padding-left: 0.625rem; padding-right: 0.625rem; overflow: hidden;">Punto di ingresso</div><div class="data" style="box-sizing: inherit; border-right: 0.0625rem solid rgb(200, 200, 200); width: 589.5px; line-height: 28px; font-size: 0.75rem; color: rgb(47, 47, 47); display: inline-block; padding-left: 0.625rem; padding-right: 0.625rem; overflow: hidden;">docker-entrypoint.sh</div></div></div><div class="link-row" style="box-sizing: inherit; position: relative; display: flex; color: rgb(47, 47, 47); font-family: Roboto, Helvetica, Arial, sans-serif, Verdana, Tahoma, 微軟正黑體; font-size: 0px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(250, 250, 250); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><div class="data-row right even" style="box-sizing: inherit; overflow: hidden; display: flex; position: relative; background-color: rgb(244, 244, 244); -webkit-box-flex: 1; flex: 1 1 0%;"><div class="data" style="box-sizing: inherit; border-right: 0.0625rem solid rgb(200, 200, 200); border-top: none !important; border-bottom: none !important; border-left: 0.0625rem solid rgb(200, 200, 200); border-image: initial !important; width: 196.5px; line-height: 28px; font-size: 0.75rem; color: rgb(47, 47, 47); display: inline-block; padding-left: 0.625rem; padding-right: 0.625rem; overflow: hidden;">Runtime</div><div class="data" style="box-sizing: inherit; border-right: 0.0625rem solid rgb(200, 200, 200); width: 589.5px; line-height: 28px; font-size: 0.75rem; color: rgb(47, 47, 47); display: inline-block; padding-left: 0.625rem; padding-right: 0.625rem; overflow: hidden;">runc</div></div></div><div class="link-row" style="box-sizing: inherit; position: relative; display: flex; color: rgb(47, 47, 47); font-family: Roboto, Helvetica, Arial, sans-serif, Verdana, Tahoma, 微軟正黑體; font-size: 0px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(250, 250, 250); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><div class="data-row right even" style="box-sizing: inherit; overflow: hidden; display: flex; position: relative; background-color: rgb(255, 255, 255); -webkit-box-flex: 1; flex: 1 1 0%;"><div class="data" style="box-sizing: inherit; border-right: 0.0625rem solid rgb(200, 200, 200); border-top: none !important; border-bottom: none !important; border-left: 0.0625rem solid rgb(200, 200, 200); border-image: initial !important; width: 196.5px; line-height: 28px; font-size: 0.75rem; color: rgb(47, 47, 47); display: inline-block; padding-left: 0.625rem; padding-right: 0.625rem; overflow: hidden;">Avvio automatico</div><div class="data" style="box-sizing: inherit; border-right: 0.0625rem solid rgb(200, 200, 200); width: 589.5px; line-height: 28px; font-size: 0.75rem; color: rgb(47, 47, 47); display: inline-block; padding-left: 0.625rem; padding-right: 0.625rem; overflow: hidden;">Enable</div></div></div><div class="link-row" style="box-sizing: inherit; position: relative; display: flex; color: rgb(47, 47, 47); font-family: Roboto, Helvetica, Arial, sans-serif, Verdana, Tahoma, 微軟正黑體; font-size: 0px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(250, 250, 250); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><div class="data-row right even" style="box-sizing: inherit; overflow: hidden; display: flex; position: relative; background-color: rgb(244, 244, 244); -webkit-box-flex: 1; flex: 1 1 0%;"><div class="data" style="box-sizing: inherit; border-right: 0.0625rem solid rgb(200, 200, 200); border-top: none !important; border-bottom: none !important; border-left: 0.0625rem solid rgb(200, 200, 200); border-image: initial !important; width: 196.5px; line-height: 28px; font-size: 0.75rem; color: rgb(47, 47, 47); display: inline-block; padding-left: 0.625rem; padding-right: 0.625rem; overflow: hidden;">CPU</div><div class="data" style="box-sizing: inherit; border-right: 0.0625rem solid rgb(200, 200, 200); width: 589.5px; line-height: 28px; font-size: 0.75rem; color: rgb(47, 47, 47); display: inline-block; padding-left: 0.625rem; padding-right: 0.625rem; overflow: hidden;">100 %</div></div></div><div class="link-row" style="box-sizing: inherit; position: relative; display: flex; color: rgb(47, 47, 47); font-family: Roboto, Helvetica, Arial, sans-serif, Verdana, Tahoma, 微軟正黑體; font-size: 0px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(250, 250, 250); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><div class="data-row right even" style="box-sizing: inherit; overflow: hidden; display: flex; position: relative; background-color: rgb(255, 255, 255); -webkit-box-flex: 1; flex: 1 1 0%;"><div class="data" style="box-sizing: inherit; border-right: 0.0625rem solid rgb(200, 200, 200); border-top: none !important; border-bottom: none !important; border-left: 0.0625rem solid rgb(200, 200, 200); border-image: initial !important; width: 196.5px; line-height: 28px; font-size: 0.75rem; color: rgb(47, 47, 47); display: inline-block; padding-left: 0.625rem; padding-right: 0.625rem; overflow: hidden;">Limite di memoria</div><div class="data" style="box-sizing: inherit; border-right: 0.0625rem solid rgb(200, 200, 200); width: 589.5px; line-height: 28px; font-size: 0.75rem; color: rgb(47, 47, 47); display: inline-block; padding-left: 0.625rem; padding-right: 0.625rem; overflow: hidden;">7868 MB</div></div></div><div class="link-row" style="box-sizing: inherit; position: relative; display: flex; color: rgb(47, 47, 47); font-family: Roboto, Helvetica, Arial, sans-serif, Verdana, Tahoma, 微軟正黑體; font-size: 0px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(250, 250, 250); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><div class="data-row right even" style="box-sizing: inherit; overflow: hidden; display: flex; position: relative; background-color: rgb(244, 244, 244); -webkit-box-flex: 1; flex: 1 1 0%;"><div class="data" style="box-sizing: inherit; border-right: 0.0625rem solid rgb(200, 200, 200); border-top: none !important; border-bottom: none !important; border-left: 0.0625rem solid rgb(200, 200, 200); border-image: initial !important; width: 196.5px; line-height: 28px; font-size: 0.75rem; color: rgb(47, 47, 47); display: inline-block; padding-left: 0.625rem; padding-right: 0.625rem; overflow: hidden;">Ambiente</div><div class="data" style="box-sizing: inherit; border-right: 0.0625rem solid rgb(200, 200, 200); width: 589.5px; line-height: 28px; font-size: 0.75rem; color: rgb(47, 47, 47); display: inline-block; padding-left: 0.625rem; padding-right: 0.625rem; overflow: hidden;">NODE_VERSION=16.14.2</div></div></div><div class="link-row" style="box-sizing: inherit; position: relative; display: flex; color: rgb(47, 47, 47); font-family: Roboto, Helvetica, Arial, sans-serif, Verdana, Tahoma, 微軟正黑體; font-size: 0px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(250, 250, 250); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><div class="data-row right even" style="box-sizing: inherit; overflow: hidden; display: flex; position: relative; background-color: rgb(255, 255, 255); -webkit-box-flex: 1; flex: 1 1 0%;"><div class="data" style="box-sizing: inherit; border-right: 0.0625rem solid rgb(200, 200, 200); border-top: none !important; border-bottom: none !important; border-left: 0.0625rem solid rgb(200, 200, 200); border-image: initial !important; width: 196.5px; line-height: 28px; font-size: 0.75rem; color: rgb(47, 47, 47); display: inline-block; padding-left: 0.625rem; padding-right: 0.625rem; overflow: hidden;"></div><div class="data" style="box-sizing: inherit; border-right: 0.0625rem solid rgb(200, 200, 200); width: 589.5px; line-height: 28px; font-size: 0.75rem; color: rgb(47, 47, 47); display: inline-block; padding-left: 0.625rem; padding-right: 0.625rem; overflow: hidden;">PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin</div></div></div><div class="link-row" style="box-sizing: inherit; position: relative; display: flex; color: rgb(47, 47, 47); font-family: Roboto, Helvetica, Arial, sans-serif, Verdana, Tahoma, 微軟正黑體; font-size: 0px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(250, 250, 250); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><div class="data-row right even" style="box-sizing: inherit; overflow: hidden; display: flex; position: relative; background-color: rgb(244, 244, 244); -webkit-box-flex: 1; flex: 1 1 0%;"><div class="data" style="box-sizing: inherit; border-right: 0.0625rem solid rgb(200, 200, 200); border-top: none !important; border-bottom: none !important; border-left: 0.0625rem solid rgb(200, 200, 200); border-image: initial !important; width: 196.5px; line-height: 28px; font-size: 0.75rem; color: rgb(47, 47, 47); display: inline-block; padding-left: 0.625rem; padding-right: 0.625rem; overflow: hidden;"></div><div class="data" style="box-sizing: inherit; border-right: 0.0625rem solid rgb(200, 200, 200); width: 589.5px; line-height: 28px; font-size: 0.75rem; color: rgb(47, 47, 47); display: inline-block; padding-left: 0.625rem; padding-right: 0.625rem; overflow: hidden;">YARN_VERSION=1.22.18</div></div></div><div class="link-row" style="box-sizing: inherit; position: relative; display: flex; color: rgb(47, 47, 47); font-family: Roboto, Helvetica, Arial, sans-serif, Verdana, Tahoma, 微軟正黑體; font-size: 0px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(250, 250, 250); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><div class="data-row right even" style="box-sizing: inherit; overflow: hidden; display: flex; position: relative; background-color: rgb(255, 255, 255); -webkit-box-flex: 1; flex: 1 1 0%;"><div class="data" style="box-sizing: inherit; border-right: 0.0625rem solid rgb(200, 200, 200); border-top: none !important; border-bottom: none !important; border-left: 0.0625rem solid rgb(200, 200, 200); border-image: initial !important; width: 196.5px; line-height: 28px; font-size: 0.75rem; color: rgb(47, 47, 47); display: inline-block; padding-left: 0.625rem; padding-right: 0.625rem; overflow: hidden;">Modalità di rete</div><div class="data" style="box-sizing: inherit; border-right: 0.0625rem solid rgb(200, 200, 200); width: 589.5px; line-height: 28px; font-size: 0.75rem; color: rgb(47, 47, 47); display: inline-block; padding-left: 0.625rem; padding-right: 0.625rem; overflow: hidden;">NAT</div></div></div><div class="link-row" style="box-sizing: inherit; position: relative; display: flex; color: rgb(47, 47, 47); font-family: Roboto, Helvetica, Arial, sans-serif, Verdana, Tahoma, 微軟正黑體; font-size: 0px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(250, 250, 250); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><div class="data-row right even" style="box-sizing: inherit; overflow: hidden; display: flex; position: relative; background-color: rgb(244, 244, 244); -webkit-box-flex: 1; flex: 1 1 0%;"><div class="data" style="box-sizing: inherit; border-right: 0.0625rem solid rgb(200, 200, 200); border-top: none !important; border-bottom: none !important; border-left: 0.0625rem solid rgb(200, 200, 200); border-image: initial !important; width: 196.5px; line-height: 28px; font-size: 0.75rem; color: rgb(47, 47, 47); display: inline-block; padding-left: 0.625rem; padding-right: 0.625rem; overflow: hidden;">Inoltro di porta</div><div class="data" style="box-sizing: inherit; border-right: 0.0625rem solid rgb(200, 200, 200); width: 589.5px; line-height: 28px; font-size: 0.75rem; color: rgb(47, 47, 47); display: inline-block; padding-left: 0.625rem; padding-right: 0.625rem; overflow: hidden;">

Host | Contenitore | Protocollo
-- | -- | --
8080 | 8080 | TCP

</div></div></div>
</body>
</html>

**Reply by snakuzzo** [reply]
La configurazione del container. Vedi il paragrafo relativo ai device `tty`.

**Reply by gmarongiu71** [reply]
Dove scusa? Sono neofito, grazie

**Reply by snakuzzo** [reply]
![aaa](https://user-images.githubusercontent.com/7205371/162681963-052c709b-1e8c-4a60-9d5c-33832eb0125f.png)

**Reply by gmarongiu71** [reply]
In teoria, come ho inserito negli screenshot, l'ho fatto

**Reply by snakuzzo** [reply]
Perdonami...ma così diventa complicato.
Facciamo ordine...Sai come funziona docker? Come funziona il mapping dei volumi e dei device ?
Una volta collegata la chiavetta al QNAP, sull'host qual è la porta corretta ? `/dev/ttyUSB0`, `/dev/ttyUSB1` o altro ?
Questo è il primo punto da smarcare.
Poi...una volta capito questo, avvia il container e prova subito ad andare sulla shell del container.

```
docker exec -it <nomecontainer> sh
```

e vedi se quella porta è presente...

```
ls -lart /dev | grep tty
```

**Reply by gmarongiu71** [reply]
Ok grazie per le dritte, stasera provo

**Reply by gmarongiu71** [reply]
Me ne da una sfilza ma nessuno con TTYUSB
crw-------    1 root     root        4,   9 Apr 11 20:34 tty9
crw-------    1 root     root        4,   8 Apr 11 20:34 tty8
crw-------    1 root     root        4,   7 Apr 11 20:34 tty7
crw-------    1 root     root        4,  63 Apr 11 20:34 tty63
crw-------    1 root     root        4,  62 Apr 11 20:34 tty62
crw-------    1 root     root        4,  61 Apr 11 20:34 tty61
crw-------    1 root     root        4,  60 Apr 11 20:34 tty60
crw-------    1 root     root        4,   6 Apr 11 20:34 tty6
crw-------    1 root     root        4,  59 Apr 11 20:34 tty59
crw-------    1 root     root        4,  58 Apr 11 20:34 tty58
crw-------    1 root     root        4,  57 Apr 11 20:34 tty57
crw-------    1 root     root        4,  56 Apr 11 20:34 tty56
crw-------    1 root     root        4,  55 Apr 11 20:34 tty55
crw-------    1 root     root        4,  54 Apr 11 20:34 tty54
crw-------    1 root     root        4,  53 Apr 11 20:34 tty53
crw-------    1 root     root        4,  52 Apr 11 20:34 tty52
crw-------    1 root     root        4,  51 Apr 11 20:34 tty51
crw-------    1 root     root        4,  50 Apr 11 20:34 tty50
crw-------    1 root     root        4,   5 Apr 11 20:34 tty5
crw-------    1 root     root        4,  49 Apr 11 20:34 tty49
crw-------    1 root     root        4,  48 Apr 11 20:34 tty48
crw-------    1 root     root        4,  47 Apr 11 20:34 tty47
crw-------    1 root     root        4,  46 Apr 11 20:34 tty46
crw-------    1 root     root        4,  45 Apr 11 20:34 tty45
crw-------    1 root     root        4,  44 Apr 11 20:34 tty44
crw-------    1 root     root        4,  43 Apr 11 20:34 tty43
crw-------    1 root     root        4,  42 Apr 11 20:34 tty42
crw-------    1 root     root        4,  41 Apr 11 20:34 tty41
crw-------    1 root     root        4,  40 Apr 11 20:34 tty40
crw-------    1 root     root        4,   4 Apr 11 20:34 tty4
crw-------    1 root     root        4,  39 Apr 11 20:34 tty39
crw-------    1 root     root        4,  38 Apr 11 20:34 tty38
crw-------    1 root     root        4,  37 Apr 11 20:34 tty37
crw-------    1 root     root        4,  36 Apr 11 20:34 tty36
crw-------    1 root     root        4,  35 Apr 11 20:34 tty35
crw-------    1 root     root        4,  34 Apr 11 20:34 tty34
crw-------    1 root     root        4,  33 Apr 11 20:34 tty33
crw-------    1 root     root        4,  32 Apr 11 20:34 tty32
crw-------    1 root     root        4,  31 Apr 11 20:34 tty31
crw-------    1 root     root        4,  30 Apr 11 20:34 tty30
crw-------    1 root     root        4,   3 Apr 11 20:34 tty3
crw-------    1 root     root        4,  29 Apr 11 20:34 tty29
crw-------    1 root     root        4,  28 Apr 11 20:34 tty28
crw-------    1 root     root        4,  27 Apr 11 20:34 tty27
crw-------    1 root     root        4,  26 Apr 11 20:34 tty26
crw-------    1 root     root        4,  25 Apr 11 20:34 tty25
crw-------    1 root     root        4,  24 Apr 11 20:34 tty24
crw-------    1 root     root        4,  23 Apr 11 20:34 tty23
crw-------    1 root     root        4,  22 Apr 11 20:34 tty22
crw-------    1 root     root        4,  21 Apr 11 20:34 tty21
crw-------    1 root     root        4,  20 Apr 11 20:34 tty20
crw--w----    1 root     root        4,   2 Apr 11 20:34 tty2
crw-------    1 root     root        4,  19 Apr 11 20:34 tty19
crw-------    1 root     root        4,  18 Apr 11 20:34 tty18
crw-------    1 root     root        4,  17 Apr 11 20:34 tty17
crw-------    1 root     root        4,  16 Apr 11 20:34 tty16
crw-------    1 root     root        4,  15 Apr 11 20:34 tty15
crw-------    1 root     root        4,  14 Apr 11 20:34 tty14
crw-------    1 root     root        4,  13 Apr 11 20:34 tty13
crw-------    1 root     root        4,  12 Apr 11 20:34 tty12
crw-------    1 root     root        4,  11 Apr 11 20:34 tty11
crw-------    1 root     root        4,  10 Apr 11 20:34 tty10
crw--w----    1 root     root        4,   1 Apr 11 20:34 tty1
crw-------    1 root     root        4,   0 Apr 11 20:34 tty0
crw-rw-rw-    1 root     root        5,   0 Apr 11 20:34 tty
crw-rw----    1 root     tty         7,  66 Apr 11 20:34 vcsu2
crw-rw----    1 root     tty         7, 130 Apr 11 20:34 vcsa2
crw-rw----    1 root     tty         7,   2 Apr 11 20:34 vcs2
crw-------    1 root     root        4,  67 Apr 11 20:34 ttyS3
crw-------    1 root     root        4,  66 Apr 11 20:34 ttyS2
crw-------    1 root     root        4,  65 Apr 11 20:34 ttyS1
crw--w----    1 root     root        4,  64 Apr 11 20:34 ttyS0
crw--w----    1 root     tty       136,   0 Apr 11 20:34 console

**Reply by snakuzzo** [reply]
Non è montata

**Reply by snakuzzo** [reply]
Devo verificare una cosa...ti tengo aggiornato

**Reply by snakuzzo** [reply]
Ma scusa...se dai lo stesso comando sull'host la vedi?
Io purtroppo non ho più il QNAP e quindi non posso provare

**Reply by gmarongiu71** [reply]
Se entro in ssh sul nas senza interrogare il docker vedo usb0 e usb1. Tra l'altro ho un UPS via USB che funziona regolarmente. Non vorrei che fosse da aggiornare il dongle.

**Reply by snakuzzo** [reply]
Non c'entra niente il dongle.
Quindi se stacchi la chiavetta vedi solo USB0 ?

**Comment by gmarongiu71** [comment]
Se stacco tutto, sia  dongle che UPS, continuo a vedere due usb, come la foto del post originale. La chiavetta si accende se la metto e quindi è alimentata. Se vuoi ci possiamo collegare vie team Viewer.

**Reply by gmarongiu71** [reply]
Per ora ho ordinato il programmatore e riproverò dopodomani dopo aver flashato. Ci sono altri consigli? Io non riesco a farlo funzionare ed ora il qnap non me lo vede più, all'inizio si.
Grazie

**Reply by snakuzzo** [reply]
Vabbè sempre più strano...comunque non credo che sia un problema di firmware.
Se riesci, contattami su telegram

**Reply by gmarongiu71** [reply]
Qua c'è scritto che potrebbe essere necessario riprogrammarlo:
https://indomus.it/componenti/cc2531-bridge-gateway-usb-zigbee/

**Reply by snakuzzo** [reply]
prova...

**Reply by gmarongiu71** [reply]
Ciao sono riuscito a risolvere montando in device della Sonoff presente sempre nella guida.
Le prove che avevo fatto con il device CC2531, senza successo, erano:
Provato ad aggiornare senza successo, preso un secondo nuovo device identico e anch'esso con o senza aggiornamento non andava.
Cambiato marca di device e funzionato semplicemente dando il RUN al Docker, che comunque per pulizia ho rimosso e configurato da capo.

**Reply by snakuzzo** [reply]
Comunque...ora che ci penso...la cc2531 solitamente non viene montata sotto `/dev/ttyUSB0`, ma sotto `/dev/ttyACM0`.
Ad ogni modo...meglio così...la CC2531 è una cacchetta in confronto a quelle con chip più recenti come la ITEAD

**Reply by gmarongiu71** [reply]
Allora puoi modificare la guida? Infatti mi compariva la ACM e non nascondo che dato i costi e l'utilizzo, per un solo device, l'avrei usata volentieri la 2531...

**Reply by snakuzzo** [reply]
secondo me al massimo si può mettere una nota spiegando che il path potrebbe essere diverso...il mount point dei device non è detto che sia esattamente come lo indichiamo. Nell'esempio leggi `USB0` perché noi abbiamo provato con una CC2530. Ma se avessi avuto altri device collegati via USB sarebbe potuto essere anche `USB3` (tanto per fare un esempio a caso)

**Reply by snakuzzo** [reply]
Noi cerchiamo di fare le guide nella maniera più precisa possibile, ma sono pur sempre guide e non dei manuali di istruzioni da seguire ad occhi chiusi. Devono anche essere uno spunto per apprendere, curiosare ed imparare cose nuove.

**Reply by gmarongiu71** [reply]
Certo assolutamente, ma mentre tra USB 0, 1, 2, ecc potrebbe essere chiaro ACM è un'altra porta. Solo un consiglio per i neofiti come me. Per il resto comunque grazie per le guide

**Reply by snakuzzo** [reply]
Vedremo come scrivere una nota...
considera comunque che la CC2531 è assolutamente sconsigliata... è un device ormai obsoleto

**Reply by gmarongiu71** [reply]
Per carità, non discuto l'obsolescenza, però se è indicata tra i testati andrebbe menzionata. Magari si possono estendere le possibili risposte in ttyusb(x) o ttyacm(x) o altro dove ovviamente la x indica il numero della periferica.
Il mio è solo un consiglio per le capre come me :D :D :D

**Reply by snakuzzo** [reply]
abbiamo aggiornato la guida

**Reply by gmarongiu71** [reply]
Grazie :)
